WooCommerce/Classes/ViewRelated/Dashboard/JetpackInstall/JetpackInstallStepsView.swift

```diff
@@ -77,26 +79,28 @@ struct JetpackInstallStepsView: View {
                 // Install steps
                 VStack(alignment: .leading, spacing: Constants.stepItemsVerticalSpacing) {
                     ForEach(JetpackInstallStep.allCases) { step in
-                        HStack(spacing: Constants.stepItemHorizontalSpacing) {
-                            if step == currentStep, step != .done {
-                                ActivityIndicator(isAnimating: .constant(true), style: .medium)
-                            } else if step > currentStep {
-                                Image(uiImage: .checkEmptyCircleImage)
-                                    .resizable()
-                                    .frame(width: Constants.stepImageSize * scale, height: Constants.stepImageSize * scale)
-                            } else {
-                                Image(uiImage: .checkCircleImage)
-                                    .resizable()
-                                    .frame(width: Constants.stepImageSize * scale, height: Constants.stepImageSize * scale)
-                            }
-
-                            Text(step.title)
-                                .font(.body)
-                                .if(step <= currentStep) {
-                                    $0.bold()
+                        viewModel.currentStep.map { currentStep in
```

Comment:
just wondering, why does each step's HStack setup have to be in the mapping block of `viewModel.currentStep`? Is it because of some issues when using `viewModel.currentStep` directly in the setup? I thought the current step can be used like:

```swift
VStack(alignment: .leading, spacing: Constants.stepItemsVerticalSpacing) {
  let currentStep = viewModel.currentStep
  ForEach(JetpackInstallStep.allCases) { step in
    ... step UI based on `currentStep`
  }
}
```

Reply:
The `.map` here is to unwrap the optional `currentStep`, since I don't feel right keeping it with a default value (`.installation`) initially. What do you think about this?

Reply:
got it! would it work if `viewModel.currentStep` is moved before `ForEach(JetpackInstallStep.allCases)` so that all step cases share the same `currentStep`?

Reply:
Right - I didn't notice the issue earlier 😅 Updated in dd7fb40db51d36de4a30bee278f655235ed0c1cd - thanks!